openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -180,3 +206,57 @@ def _email_url_parser(url_name, extra_param=None):
     dashboard_url_path = reverse(url_name) + extra_param if extra_param else reverse(url_name)
     dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
     return urlparse.urlunparse(dashboard_link_parts)
+
+
+def get_credit_providers_by_course(course_key):
+    """Get the course information from ecommerce and parse the data to get providers.
+
+    Arguments:
+        course_key (CourseKey): The identifier for the course.
+
+    Returns:
+        list, containing course providers.
+    """
+    credit_config = CreditConfig.current()
+
+    providers_data = None
+    cache_key = None
+    if credit_config.is_cache_enabled:
+        cache_key = '{key_prefix}.{course_key}'.format(
+            key_prefix=credit_config.CACHE_KEY, course_key=unicode(course_key)
+        )
+        providers_data = cache.get(cache_key)
+
+    if providers_data is not None:
```

Comment:
@rlucioni If providers_name does not find any provider than it will set the cache with [] for specific time. And if we do following 
```
if providers_data:
```
it will return false again try to hit the ecommerce. 
That's i did this.
```
if providers_data is not None:
```
It will return the True and return the empty list unless cache expire.